src/pybind/mgr/volumes/fs/volume.py

```diff
@@ -864,6 +885,25 @@ def pin_subvolume_group(self, **kwargs):
             ret = self.volume_exception_to_retval(ve)
         return ret
 
+    def subvolume_group_exists(self, **kwargs):
+        volname = kwargs['vol_name']
+        ret = 0, "", ""
+        volume_exists = False
+
+        try:
+            with open_volume(self, volname) as fs_handle:
+                volume_exists = True
+                res = has_subdir(fs_handle, self.volspec.base_dir, filter_entries=[
+                                 dir.encode('utf-8') for dir in self.volspec.INTERNAL_DIRS])
+                if res:
+                    ret = 0, "Subvolumegroup(s) exist", ""
+                else:
+                    ret = 1, "No subvolumegroup exist", ""
+        except VolumeException as ve:
+            if not ve.errno == -errno.ENOENT or not volume_exists:
```

Comment:
What about volumes exists but the path '/volumes' doesn't exist ? In this case, this is returning success.

Reply:
I handled this case as it is done in other places.Should I handle it separately?

Reply:
I think since subvolumegroup/subvolume is not present, we should return success with success string

Reply:
Let's get rid of opendir at line no. 897 and rely on ENOENT as below.
```
if volume_exists and ve.errno == -errno.ENONET:
    ret = 0, "no subvolumegroup exists", ""
else:
    ret = self.volume_exception_to_retval(ve)
```

Reply:
Done!!